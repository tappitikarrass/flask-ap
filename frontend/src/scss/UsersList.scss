@use "base16" as *;

#users-list.content {
    display: flex;
    flex-direction: column;
}

// desktop
@media screen and (min-width: 810px) {
    #users-list.content {
        width: 90vw;
        margin-left: 5vw;
    }
}

// tablet
@media screen and (min-width: 500px) and (max-width: 810px) {
    #users-list.content {
        width: 100vw;
        margin-left: 0vw;
    }
}

// phone
@media screen and (max-width: 500px) {
    #users-list.content {
        width: 100vw;
        margin-left: 0;
    }
}
