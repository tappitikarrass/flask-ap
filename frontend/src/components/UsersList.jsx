import React, { useEffect, useState } from 'react';
import { useCookies } from 'react-cookie';
import { useNavigate } from 'react-router-dom';
import '../scss/UsersList.scss';

function UsersList() {
  const navigate = useNavigate();
  const [cookies] = useCookies('token', 'username');
  const [users, setUsers] = useState([]);
  const [reqStatus, setReqStatus] = useState();

  useEffect(() => {
    async function effect() {
      const response = await fetch(
        'http://localhost/backend/user',
        {
          method: 'GET',
          headers: {
            Authorization: `Bearer ${cookies.token}`,
          },
        },
      );
      /* istanbul ignore next */
      const json = await response.json();
      /* istanbul ignore next */
      setUsers(json);
      /* istanbul ignore next */
      setReqStatus(response.status);
    }
    /* istanbul ignore if */
    if (cookies.token === 'undefined') {
      navigate('/login');
    } else {
      effect();
    }
  }, []);

  /* istanbul ignore if */
  if (reqStatus === 200) {
    return (
      <div id="users-list" className="content">
        <table id="users-table">
          <tbody>
            <tr>
              <td>id</td>
              <td>username</td>
              <td>email</td>
              <td>Fistname</td>
              <td>Lastname</td>
              <td>Phone</td>
            </tr>
            {
              users.map((value) => (
                <tr>
                  <td>{value.user_id}</td>
                  <td>{value.username}</td>
                  <td>{value.email}</td>
                  <td>{value.firstname}</td>
                  <td>{value.lastname}</td>
                  <td>{value.phone}</td>
                </tr>
              ))
            }
          </tbody>
        </table>
      </div>
    );
  }
  /* istanbul ignore if */
  if (reqStatus !== 200) {
    return (
      <div id="users-list" className="content">
        <h2>Not allowed!</h2>
      </div>
    );
  }
}

export default UsersList;
