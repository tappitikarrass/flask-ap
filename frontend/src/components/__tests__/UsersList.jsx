import React from "react";
import { HashRouter } from "react-router-dom";
import { create, act } from "react-test-renderer";
import UsersList from '../UsersList';

describe('UsersList', () => {
  it('',() => {
    let component;
    act(() => {
      component = create(
        <HashRouter>
          <UsersList />
        </HashRouter>
      );
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
