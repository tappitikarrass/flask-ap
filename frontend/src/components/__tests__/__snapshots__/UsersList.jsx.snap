// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList  1`] = `
<div
  className="content"
  id="users-list"
>
  <h2>
    Not allowed!
  </h2>
</div>
`;
